### PR TITLE
Add online backup.

### PR DESCRIPTION
**Pull request by AlexanderTaeschner**
Addresses #17

### CODE REVIEW

File: src/Microsoft.Data.Sqlite.Core/SqliteConnection.cs [hunk 1]
```diff
@@ -380,6 +380,42 @@ public virtual void EnableExtensions(bool enable = true)
...
+            if (backup.ptr == IntPtr.Zero)
+            {
+                SqliteException.ThrowExceptionForDatabaseError(_db);
+            }
+
+            var rc = raw.sqlite3_backup_step(backup, -1);
+            SqliteException.ThrowExceptionForRC(rc, _db);
+
+            rc = raw.sqlite3_backup_finish(backup);
+            SqliteException.ThrowExceptionForRC(rc, _db);
```
Comment (bricelam) [inline src/Microsoft.Data.Sqlite.Core/SqliteConnection.cs]: No need to call `sqlite3_backup_finish`, just wrap `backup` inside a `using` statement.

File: src/Microsoft.Data.Sqlite.Core/SqliteConnection.cs [hunk 2]
```diff
@@ -380,6 +380,42 @@ public virtual void EnableExtensions(bool enable = true)
...
+                || destination.Handle == null
+                || destination.Handle.ptr == IntPtr.Zero)
+            {
+                throw new InvalidOperationException(Resources.CallRequiresOpenConnection(nameof(BackupDatabase)));
+            }
+
+            var backup = raw.sqlite3_backup_init(destination.Handle, destinationName, _db, sourceName);
+            if (backup.ptr == IntPtr.Zero)
+            {
+                SqliteException.ThrowExceptionForDatabaseError(_db);
```
Comment (bricelam) [inline src/Microsoft.Data.Sqlite.Core/SqliteConnection.cs]: I wonder if we should be doing this elsewhere...

File: src/Microsoft.Data.Sqlite.Core/SqliteConnection.cs (hunk 1 above)
Comment (AlexanderTaeschner) [reply src/Microsoft.Data.Sqlite.Core/SqliteConnection.cs]: Okay, changed.

File: src/Microsoft.Data.Sqlite.Core/SqliteConnection.cs [hunk 3]
```diff
@@ -380,6 +380,41 @@ public virtual void EnableExtensions(bool enable = true)
...
+
+            using (var backup = raw.sqlite3_backup_init(destination.Handle, destinationName, _db, sourceName))
+            {
+                if (backup.ptr == IntPtr.Zero)
+                {
+                    SqliteException.ThrowExceptionForDatabaseError(_db);
+                }
+
+                var rc = raw.sqlite3_backup_step(backup, -1);
+                SqliteException.ThrowExceptionForRC(rc, _db);
```
Comment (bricelam) [inline src/Microsoft.Data.Sqlite.Core/SqliteConnection.cs]: The documentation doesn't say, but it looks like (after a lot of digging) this is supposed to be `destincation._db` too. I'll fix.

File: src/Microsoft.Data.Sqlite.Core/SqliteConnection.cs (hunk 2 above)
Comment (bricelam) [reply src/Microsoft.Data.Sqlite.Core/SqliteConnection.cs]: (I checked and couldn't find any.)

**Pull request merged**